packages/api-elements/lib/define-value-of.js

```diff
@@ -24,20 +25,35 @@ const {
   getStructureMembers,
 } = require('./utils');
 
+/**
+ * @typedef InheritedAttrs
+ * @type {object}
+ * @property {boolean} isFixed
+ * @property {boolean} isFixedType
+ * @property {boolean} isNullable
+ */
+
 /**
  * Map the element values
  * @param {element} e - element
+ * @param {InheritedAttrs} inheritedAttrs - inherited attributes
  * @param {function} f - map function
  * @param {object=} elements - object map of elements to look for inherited types
  * @return {any}
  */
-function mapValue(e, f, elements) {
+function mapValue(e, f, inheritedAttrs, elements) {
   if (e === undefined) {
     return undefined;
   }
 
+  const attrs = {
+    isFixed: inheritedAttrs.isFixed || isFixed(e),
+    isFixedType: (inheritedAttrs.isFixed && isPrimitive(e)) || isFixedType(e),
```

Comment:
Fixed-type isn't an inherited type attribute, I think this part should be removed?

Reply:
Indeed. Addressed in ef0df9c.

I'm now realizing it's missing tests for [required](https://github.com/apiaryio/api-elements.js/pull/564/files#diff-ed3f4907fe76fdf2e90227edd1bba42cR158).